app/src/main/java/com/example/plantsapp/MainActivity.kt

```diff
@@ -2,12 +2,40 @@ package com.example.plantsapp
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.fragment.app.Fragment
+import com.example.plantsapp.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity() {
 
+    private lateinit var binding: ActivityMainBinding
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.bottomNavigation.setOnItemSelectedListener { menuItem ->
+            when (menuItem.itemId) {
+
+                R.id.bottom_nav_tasks -> {
+                    openFragment(TasksFragment())
+                    true
+                }
+
+                R.id.bottom_nav_plants -> {
+                    openFragment(PlantsFragment())
+                    true
+                }
+
+                else -> true
+            }
+        }
```

Comment:
2. Each branch of when return true , so maybe  you can return true at the end and leave 
```
when (menuItem.itemId) {
                  R.id.bottom_nav_tasks -> openFragment(TasksFragment())
                  R.id.bottom_nav_plants -> openFragment(PlantsFragment())
              }
true 
```